Sources/StreamChat/Repositories/PollsRepository.swift

```diff
@@ -0,0 +1,269 @@
+                }
+            } else {
+                completion?(error)
+            }
+        }
+    }
+    
+    func closePoll(
+        pollId: String,
+        completion: ((Error?) -> Void)? = nil
+    ) {
+        let request = UpdatePollPartialRequestBody(
+            pollId: pollId,
+            set: ["is_closed": .bool(true)]
+        )
+        apiClient.request(
+            endpoint: .updatePollPartial(pollId: pollId, updatePollPartialRequest: request)
+        ) {
+            completion?($0.error)
+        }
+    }
+    
+    func suggestPollOption(
+        pollId: String,
+        text: String,
+        position: Int? = nil,
+        custom: [String: RawJSON]? = nil,
+        completion: ((Error?) -> Void)? = nil
+    ) {
+        let request = CreatePollOptionRequestBody(
+            pollId: pollId,
+            text: text,
+            position: position,
+            custom: custom
+        )
+        apiClient.request(
+            endpoint: .createPollOption(pollId: pollId, createPollOptionRequest: request),
+            completion: {
+                completion?($0.error)
+            }
+        )
+    }
+    
+    func queryPollVotes(
+        query: PollVoteListQuery,
+        completion: ((Result<[PollVote], Error>) -> Void)? = nil
+    ) {
+        apiClient.request(
+            endpoint: .queryPollVotes(pollId: query.pollId, query: query)
+        ) { [weak self] (result: Result<PollVoteListResponse, Error>) in
+            switch result {
+            case let .success(payload):
+                var votes: [PollVote] = []
+                self?.database.write({ session in
+                    votes = try session.savePollVotes(payload: payload, query: query, cache: nil).map { try $0.asModel() }
+                }, completion: { error in
+                    if let error = error {
+                        completion?(.failure(error))
+                    } else {
+                        completion?(.success(votes))
+                    }
+                })
+            case let .failure(error):
+                completion?(.failure(error))
+            }
+        }
+    }
+    
+    func queryPollVotes(
+        pollId: String,
+        limit: Int?,
+        next: String?,
+        prev: String?,
+        sort: [SortParamRequest?],
+        filter: [String: RawJSON]?,
+        completion: ((Result<PollVoteListResponse, Error>) -> Void)? = nil
+    ) {
+        let request = QueryPollVotesRequestBody(
+            pollId: pollId,
+            limit: limit,
+            next: next,
+            prev: prev,
+            sort: sort,
+            filter: filter
+        )
+        apiClient.request(
+            endpoint: .queryPollVotes(pollId: pollId, queryPollVotesRequest: request),
+            completion: { [weak self] result in
+                guard let self else { return }
+                switch result {
+                case let .success(response):
+                    database.write { session in
```

Comment:
Fix compilation issue:
```suggestion
                    self.database.write { session in
```